### DISCUSSION TITLE
How to check OPA policy without commenting?

### DISCUSSION THREAD

**Discussion by gtirloni** [infracost/infracost#1665] [Q&A]
It seems we can only pass a OPA policy file to the `infracost comment` command, is that right?

I would just like to check the output of `infracost` in our CI/CD pipeline to see if it passed our cost policies or not before deciding what to do about it.

**Comment by alikhajeh1** [comment]
@gtirloni we use to have an example but we removed it when we released infracost comment: https://github.com/infracost/actions/tree/v1.1.0/examples/opa
Does that help? So you can install OPA and run that, then based on that output decide what to do next. (we created https://github.com/open-policy-agent/setup-opa to install OPA in github actions but it's an executable that can be installed in any CI/CD too).

**Reply by gtirloni** [reply]
I think I re-discovered that example by trial and error :-)

We integrate infracost with Atlantis. One supports OPA directly, the latter through conftest.

I had to add two slightly different policies to our repository to satisfy both.

Policy for infracost:
```rego
package infracost

deny[out] {
  maxDiff = 50.0

  msg := sprintf(
    "Total monthly cost diff must be less than $%.2f (actual diff is $%v)",
    [maxDiff, to_number(input.diffTotalMonthlyCost)],
  )

  out := {
    "msg": msg,
    "failed": to_number(input.diffTotalMonthlyCost) >= maxDiff
  }
}
```

Policy for Atlantis/conftest:

```
package infracost

deny[msg] {
  maxDiff = 50.0

  to_number(input.diffTotalMonthlyCost) >= maxDiff

  msg := sprintf(
    "Total monthly cost diff must be less than $%v (actual diff is $%v)",
    [maxDiff, to_number(input.diffTotalMonthlyCost)],
  )
}
```

I was hoping I could have a single policy file that could be used with both, but Atlantis' policy checks work exclusively with conftest and I had to satisfy that one too.

When I posted this discussion, I was still trying to get an `exit code = 1` from infracost when evaluating a policy, so I could plug that into Atlantis and make the plan fail. But that wouldn't be a good solution because, through native policy checks, Atlantis can work with "policy owners" (users that can override policy check failures) and we needed that.

**Reply by alikhajeh1** [reply]
@gtirloni we created specific examples for Atlantis with conftest: https://github.com/infracost/infracost-atlantis#additional-examples did those work? PRs are also appreciated if the [readme or examples](https://github.com/infracost/infracost-atlantis/blob/master/examples/conftest/README.md) need changes 🚀
